### PR TITLE
[10.0][FIX] website_sale_charge_payment_fee: Use a browse record if there a…

### PR DESCRIPTION
**Pull request by SimoRubi**
…re no acquirers

### CODE REVIEW

**Comment by SimoRubi** [comment]
@OCA/e-commerce-maintainers can someone please take a look? Thanks

**Comment by SimoRubi** [comment]
@OCA/e-commerce-maintainers can someone please take a look?
How can I help this PR to be merged?
Thanks

File: website_sale_charge_payment_fee/controllers/website_sale.py
Comment (simahawk) [inline website_sale_charge_payment_fee/controllers/website_sale.py:30]: I don't know this module but I'm a bit sceptic about this change. The original module provides a list of rendered HTML, hence what you would have here is not an ID or a recordset and the template does seem to take this into account.
Am I wrong?

See https://github.com/odoo/odoo/blob/10.0/addons/website_sale/controllers/main.py#L747

**Review by simahawk — CHANGES_REQUESTED** [review]

File: website_sale_charge_payment_fee/controllers/website_sale.py
Comment (SimoRubi) [reply website_sale_charge_payment_fee/controllers/website_sale.py:30]: Hi, thanks for checking!
Actually the original module provides a list of records found with `search` https://github.com/odoo/odoo/blob/10.0/addons/website_sale/controllers/main.py#L744-L746
and then appended one by one to `value['acquirers]` in https://github.com/odoo/odoo/blob/10.0/addons/website_sale/controllers/main.py#L760; the rendered HTML is just assigned to the `button` field of every acquirer in the list.

Comment (simahawk) [reply website_sale_charge_payment_fee/controllers/website_sale.py:30]: hmm seems you are right. Sorry, I probably checked too quickly :sweat: . Then I don't see any blockers...

**Review by simahawk — APPROVED** [review]
reminder: version update is not needed any more ;)

**Comment by SimoRubi** [comment]
@OCA/e-commerce-maintainers how can I help this PR to be merged? Thanks

**Comment by simahawk** [comment]
@SimoRubi tests are failing there, can you check?

**Comment by SimoRubi** [comment]
@simahawk rebased on `10.0`.
Tests are failing because they are already failing in `10.0`:
image
(from https://travis-ci.org/OCA/e-commerce/branches)

**Comment by SimoRubi** [comment]
@simahawk does this PR have to wait for the fix of `10.0`?
If no-one is tackling it, I can at least try.

**Comment by simahawk** [comment]
@SimoRubi 10.0 is green https://github.com/OCA/e-commerce/commits/10.0

image

What you see there is an old build. The green one is pretty old and might fail today.

In your build I see `raise UserError(_("Unable to find Wkhtmltopdf on this system. The PDF can not be created."))`.

You can rebase on 10.0 and see if it happens again.

**Comment by SimoRubi** [comment]
@simahawk the last commit is _Unverified_ because it has [ci skip]:
image
That means that Travis did not run on that commit, you can verify that by clicking on the :heavy_check_mark: :
image

The last _verified_ commit is the previous one, that is red:
image
and its travis build corresponds to the latest build of branch `10.0` that is https://travis-ci.org/OCA/e-commerce/builds/564785979 and is failed.

Anyway I already rebased like I said in https://github.com/OCA/e-commerce/pull/264#issuecomment-583364617 and from that moment on, tests began failing in this PR too (they were green before, when you first tried to merge).

**Comment by simahawk** [comment]
@SimoRubi oh yeah, sorry (repeat with me: don't do 3 things at the same time when answering to ppl :sweat_smile: )
Then... you know what to do: try to fix 10.0 :)

**Comment by SimoRubi** [comment]
> @SimoRubi oh yeah, sorry (repeat with me: don't do 3 things at the same time when answering to ppl  )
> Then... you know what to do: try to fix 10.0 :)

@simahawk I've fixed `10.0` in https://github.com/OCA/e-commerce/pull/366 and rebased, can you try again to merge?
Thanks!

**Pull request merged**
